ocrd/model/ocrd_mets.py

```diff
@@ -99,32 +103,44 @@ def find_files(self, ID=None, fileGrp=None, groupId=None, mimetype=None, local_o
         file_clause = ''
         if ID is not None:
             file_clause += '[@ID="%s"]' % ID
-        if groupId is not None:
-            file_clause += '[@GROUPID="%s"]' % groupId
         if mimetype is not None:
             file_clause += '[@MIMETYPE="%s"]' % mimetype
         # TODO lxml says invalid predicate. I disagree
         #  if local_only:
         #      file_clause += "[mets:FLocat[starts-with(@xlink:href, 'file://')]]"
 
-        file_els = self._tree.getroot().findall(".//mets:fileGrp%s/mets:file%s" % (fileGrp_clause, file_clause), NS)
-        for el in file_els:
-            file_id = el.get('ID')
+        # Search
+        file_ids = self._tree.getroot().xpath("//mets:fileGrp%s/mets:file%s/@ID" % (fileGrp_clause, file_clause), namespaces=NS)
+        if pageId is not None:
+            by_pageid = self._tree.getroot().xpath('//mets:div[@TYPE="page"][@ID="%s"]/mets:fptr/@FILEID' % pageId, namespaces=NS)
+            if by_pageid:
+                file_ids = [i for i in by_pageid if i in file_ids]
+
+        # instantiate / get from cache
+        for file_id in file_ids:
+            el = self._tree.getroot().find('.//mets:file[@ID="%s"]' % file_id, NS)
             if file_id not in self._file_by_id:
-                self._file_by_id[file_id] = OcrdFile(el, baseurl=self.baseurl)
+                self._file_by_id[file_id] = OcrdFile(el, baseurl=self.baseurl, mets=self)
             if local_only:
                 url = el.find('mets:FLocat', NS).get('{%s}href' % NS['xlink'])
-                if not url.startswith('file://'):
+                if not (url.startswith('file://') or '://' not in url):
```

Comment:
First condition is a special case of the second and thus superfluous.

Reply:
The condition checks whether the URL is either a `file://` URL or not a URL at all (i.e. a relative path). If it is neither, skip the file.

I'll add a comment to make that clearer.